src/components/menu/menu.ts

```diff
@@ -251,25 +338,109 @@ export class Menu extends Component<Props, SpreadsheetChildEnv> {
     this.subMenu.parentMenu = undefined;
   }
 
-  onClickMenu(menu: Action, menuIndex: number, ev: MouseEvent) {
+  onClickMenu(menu: Action, ev: MouseEvent) {
     if (this.isEnabled(menu)) {
       if (this.isRoot(menu)) {
-        this.openSubMenu(menu, menuIndex, ev);
+        this.openSubMenuViaMouse(menu, ev);
       } else {
         this.activateMenu(menu);
       }
     }
   }
 
-  onMouseOver(menu: Action, position: Pixel, ev: MouseEvent) {
+  onMouseMove(menu: Action, ev: MouseEvent) {
+    this.subMenu.shouldSelectFirstItem = false;
+    this.subMenu.menuIndex = this.clickableMenuItems.findIndex(
+      (menuItem) => menuItem.id === menu.id
+    );
     if (this.isEnabled(menu)) {
       if (this.isRoot(menu)) {
-        this.openSubMenu(menu, position, ev);
+        this.openSubMenuViaMouse(menu, ev);
       } else {
         this.closeSubMenu();
       }
     }
   }
+
+  onKeydown(ev: KeyboardEvent) {
+    ev.preventDefault();
+    ev.stopPropagation();
+    if (ev.key.startsWith("Arrow")) {
+      this.processArrows(ev);
+    } else if (ev.key === "Enter") {
+      if (this.subMenu.menuIndex === undefined) return;
+      const menu = this.clickableMenuItems[this.subMenu.menuIndex];
+      if (this.isEnabled(menu)) {
```

Comment:
@hokolomopo `this.isEnabled` also check whether the menu is read-only allowed, so it's not only `menu.isEnabled()`. Deleting it would make one test in `context_menu` fail.